tests/integration-tests/tests/health_checks/test_gpu_health_checks.py

```diff
@@ -0,0 +1,193 @@
+    remote_command_executor = RemoteCommandExecutor(cluster)
+    slurm_commands = scheduler_commands_factory(remote_command_executor)
+
+    # Submit job to the test nodes
+    queue_cr_expected_nodes_health_statuses = expected_nodes_health_statuses.items()
+    for queue, cr_expected_nodes_health_statuses in queue_cr_expected_nodes_health_statuses:
+        no_of_nodes = len(cr_expected_nodes_health_statuses.keys())
+        job_id = slurm_commands.submit_command_and_assert_job_accepted(
+            submit_command_args={
+                "command": "srun sleep 1",
+                "host": ",".join(
+                    node_health_status.node_name for cr, node_health_status in cr_expected_nodes_health_statuses.items()
+                ),
+                "partition": queue,
+                "slots": no_of_nodes,
+                "nodes": no_of_nodes,
+            }
+        )
+        for node_health_status in cr_expected_nodes_health_statuses.values():
+            node_health_status.latest_job = job_id
+
+    # Wait for all jobs to be completed
+    slurm_commands.wait_job_queue_empty()
+
+    # Check if GPU Health Checks Manager was started on all nodes and actual Health Checks executed for nodes where
+    # its enabled and the instance type is GPU-enabled.
+    for _, cr_expected_nodes_health_statuses in queue_cr_expected_nodes_health_statuses:
+        for node_health_status in cr_expected_nodes_health_statuses.values():
+            node_address = slurm_commands.get_node_addr(node_name=node_health_status.node_name)
+            _assert_file_content_in_compute_node(
+                HEALTH_CHECK_LOG_FILE,
+                node_address,
+                cluster,
+                [rf".*JobID {node_health_status.latest_job}.*Running GPU Health Check with DCGMI.*"],
+                should_exist=node_health_status.health_check_executed,
+            )
+
+    # Simulate failing GPU Health Check and assert the node is set to DRAIN
+    _test_failing_gpu_health_checks(
+        slurm_commands=slurm_commands,
+        cluster=cluster,
+        target_node=expected_nodes_health_statuses["queue-1"]["compute-resource-3"],
+        target_queue="queue-1",
+        failure_script_path=test_datadir / "mock_failing_gpu_health_check.sh",
+        successful_script_path=test_datadir / "mock_successful_gpu_health_check.sh",
+        rollback_script_path=test_datadir / "restore_gpu_health_check.sh",
+    )
+
+
+def _test_failing_gpu_health_checks(
+    slurm_commands,
+    cluster,
+    target_node,
+    target_queue,
+    failure_script_path,
+    successful_script_path,
+    rollback_script_path,
+):
+    node_address = slurm_commands.get_node_addr(node_name=target_node.node_name)
+    compute_node_remote_command_executor = RemoteCommandExecutor(cluster, compute_node_ip=node_address)
+
+    # Mock failing GPU Health Checks
+    results_from_compute_node = compute_node_remote_command_executor.run_remote_script(failure_script_path).stdout
```

Comment:
the mock and unmock can be done in the headnode. Doing this, it would probably simplify the code (e.g. no need to get compute ip address)